packages/components/src/DataList/components/DataListLayoutInternal/DataListHeader.tsx

```diff
@@ -0,0 +1,55 @@
+import React from "react";
+import { DataListLayoutInternal } from "./DataListLayoutInternal";
+import { Breakpoints } from "../../DataList.const";
+import styles from "../../DataList.css";
+import {
+  DataListHeader as DataListHeaderType,
+  DataListItemTypeFromHeader,
+  DataListObject,
+  DataListProps,
+} from "../../DataList.types";
+import { sortSizeProp } from "../../DataList.utils";
+import { DataListLayoutProps } from "../DataListLayout/DataListLayout";
+
+interface DataListHeaderProps<T extends DataListObject> {
+  layouts: React.ReactElement<DataListLayoutProps<T>>[] | undefined;
+  mediaMatches?: Record<Breakpoints, boolean>;
+  headerData?: DataListItemTypeFromHeader<T, DataListHeaderType<T>>;
+  headerVisibility: NonNullable<DataListProps<T>["headerVisibility"]>;
```

Comment:
nit: consisitency. same goes for the other ones

```suggestion
  readonly layouts: React.ReactElement<DataListLayoutProps<T>>[] | undefined;
  readonly mediaMatches?: Record<Breakpoints, boolean>;
  readonly headerData?: DataListItemTypeFromHeader<T, DataListHeaderType<T>>;
  readonly headerVisibility: NonNullable<DataListProps<T>["headerVisibility"]>;
```